### PR TITLE
Smooth pixel screens shakes!

### LINKED ISSUE #4870: Being inside a locker during meteors allows you to see into adjacent rooms.
What happened:

- I was an assistant hiding in the locker in a bridge by the captain's office (yes I know...).
- Meaty Ores have been detected on a collision course with the station.
- The station starts doing the impact animations from meteors. The display kept jerking, like usual, but when it did so it also allowed me to see into the captain's quarters, even though there was a wall (and a locker) in the way and no reason for that to happen.
- This for some reason did not seem to happen when the locker was open.

Why this is bad:

- Could be used as an exploit, although admittedly in very edge cases.

### PR DESCRIPTION
**Pull request by MrsTonedOne**
This pr brought to you by 38 plays of "Shake it off"

fixes #4870 

:cl:
fix: screen shakes will no longer allow you to see through walls
tweak: screen shakes will now move with you when moving rather than lag your client behind your mob
tweak: screen shakes are now less laggy in higher tickrates
/:cl:

Demo: both recorded on a server running at 16fps. (the new one looks cooler on higher fps while the old one just looks faster to move while keeping the same time between changes, so it looks odd)
https://tgstation13.org/msoshit/shakeitoff/old.flv
https://tgstation13.org/msoshit/shakeitoff/new.flv

### CODE REVIEW

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -187,20 +187,26 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
...
+	var/min = 0-((strength+1)*world.icon_size)
+
+	for(var/i=0, i<duration, i++)
+		var/newx = rand(min,max)
+		var/newy = rand(min,max)
+		if (i == 0)
+			animate(C, pixel_x=newx, pixel_y=newy, 1, 1, ELASTIC_EASING)
+		else
+			animate(pixel_x=newx, pixel_y=newy, 1, 1, ELASTIC_EASING)
+		sleep 1
```
Comment (MrStonedOne) [inline code/modules/mob/mob_helpers.dm]: This sleep brought to you by byond http://www.byond.com/forum/?post=2001341

**Comment by Xhuis** [comment]
Could you put a .gif in the OP?

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (duncathan) [inline code/modules/mob/mob_helpers.dm]: can you put the 1 between parentheses? It's tickling my autism like this

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -187,20 +187,26 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
...
+	var/min = 0-((strength+1)*world.icon_size)
+
+	for(var/i=0, i<duration, i++)
+		var/newx = rand(min,max)
+		var/newy = rand(min,max)
+		if (i == 0)
+			animate(C, pixel_x=newx, pixel_y=newy, 1, 1, ELASTIC_EASING)
+		else
+			animate(pixel_x=newx, pixel_y=newy, 1, 1, ELASTIC_EASING)
+		sleep(1)
```
Comment (MrStonedOne) [inline code/modules/mob/mob_helpers.dm]: This sleep brought to you by byond http://www.byond.com/forum/?post=2001341

File: code/modules/mob/mob_helpers.dm [hunk 3]
```diff
@@ -187,20 +187,26 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
...
+	if(!M || !M.client || M.shakecamera)
+		return
+	M.shakecamera = 1
+	var/client/C = M.client
+	var/oldx = C.pixel_x
+	var/oldy = C.pixel_y
+	var/max = (strength+1)*world.icon_size
+	var/min = 0-((strength+1)*world.icon_size)
+
+	for(var/i=0, i<duration, i++)
```
Comment (RemieRichards) [inline code/modules/mob/mob_helpers.dm]: ```for(var/i in 0 to duration-1)``` pls

**Comment by AnturK** [comment]
Fixes #4870 probably.

**Comment by MrStonedOne** [comment]
Updated with videos

(gifs can't really capture this)

**Pull request merged**